packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+		EventType.AllUserCodesDeleted,
+		EventType.MasterCodeChanged,
+		EventType.UserCodeChanged,
+		EventType.LockReset,
+		EventType.ConfigurationChanged,
+		EventType.LowBattery,
+		EventType.NewBattery,
+		EventType.Unknown,
+	];
+
+	if (!valueDB.hasMetadata(recordEventTypeID)) {
+		valueDB.setMetadata(recordEventTypeID, {
+			...ValueMetadata.ReadOnlyString,
+			label: `record event type`,
+			description: "Event Type for latest logging record",
+			states: enumValuesToMetadataStates(EventType, supportedEventTypes),
+		});
+	}
+	if (
+		!valueDB.hasMetadata(recordNumberID) &&
+		typeof supportedRecordsNumber != "undefined"
+	) {
+		valueDB.setMetadata(recordNumberID, {
+			...ValueMetadata.ReadOnlyNumber,
+			min: 1,
+			max: supportedRecordsNumber,
+			label: `record number`,
+			description: "Record Number for latest logging record",
+		});
+	}
+	if (!valueDB.hasMetadata(recordUserIDID)) {
+		valueDB.setMetadata(recordUserIDID, {
+			...ValueMetadata.ReadOnlyNumber,
+			min: 0,
+			label: `record user id`,
+			description: "User ID for latest logging record",
+		});
+	}
+
+	const codeMetadata: ValueMetadata = {
+		...(Buffer.isBuffer(userCode)
+			? ValueMetadata.Buffer
+			: ValueMetadata.String),
+		minLength: 4,
+		maxLength: 10,
+		label: `record user code`,
+	};
+	if (valueDB.getMetadata(recordUserCodeID)?.type !== codeMetadata.type) {
+		valueDB.setMetadata(recordUserCodeID, codeMetadata);
+	}
+}
+
+@API(CommandClasses["Door Lock Logging"])
+export class DoorLockLoggingCCAPI extends PhysicalCCAPI {
+	public supportsCommand(cmd: DoorLockLoggingCommand): Maybe<boolean> {
+		switch (cmd) {
+			case DoorLockLoggingCommand.RecordsSupportedGet:
+			case DoorLockLoggingCommand.RecordsSupportedReport:
+			case DoorLockLoggingCommand.RecordGet:
+			case DoorLockLoggingCommand.RecordReport:
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "supportedRecordsNumber":
+				return (await this.getSupportedRecordsNumber())?.[property];
+			case "record":
+				// Fetches latest record by default
+				return (await this.getRecord(LATEST_RECORD_NUMBER_KEY))?.[
+					property
+				];
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	public async getSupportedRecordsNumber(): Promise<any> {
+		this.assertSupportsCommand(
+			DoorLockLoggingCommand,
+			DoorLockLoggingCommand.RecordsSupportedGet,
+		);
+
+		const cc = new DoorLockLoggingCCRecordsSupportedGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response =
+			await this.driver.sendCommand<DoorLockLoggingCCRecordsSupportedReport>(
+				cc,
+				this.commandOptions,
+			);
+		if (response) {
+			return pick(response, ["supportedRecordsNumber"]);
+		}
+	}
+
+	public async getRecord(
+		recordNumber: number = LATEST_RECORD_NUMBER_KEY,
+	): Promise<any> {
```

Comment:
```suggestion
	): Promise<Record | undefined> {
```

Reply:
I tried that before, and the house of cards falls down when I do that. I tried tinkering with the types all about, and I can't seem to get it working.

Reply:
The problem was that `pick` returns an object with the picked property. You have to access the desired property directly.